yarn/src/main/scala/org/apache/spark/deploy/yarn/security/CredentialUpdater.scala

```diff
@@ -41,16 +43,18 @@ private[spark] class ExecutorDelegationTokenUpdater(
     SparkHadoopUtil.get.getConfBypassingFSCache(
       hadoopConf, new Path(credentialsFile).toUri.getScheme)
 
-  private val delegationTokenRenewer =
+  private val credentialUpdater =
     Executors.newSingleThreadScheduledExecutor(
-      ThreadUtils.namedThreadFactory("Delegation Token Refresh Thread"))
+      ThreadUtils.namedThreadFactory("Credential Refresh Thread"))
 
-  // On the executor, this thread wakes up and picks up new tokens from HDFS, if any.
-  private val executorUpdaterRunnable =
+  // This thread wakes up and picks up new credentials from HDFS, if any.
+  private val credentialUpdaterRunnable =
     new Runnable {
       override def run(): Unit = Utils.logUncaughtExceptions(updateCredentialsIfRequired())
     }
 
+  @volatile private var timeOfNextUpdate = sparkConf.get(CREDENTIALS_UPDATE_TIME)
```

Comment:
This is only ever used inside `updateCredentialsIfRequired`, do you need to keep it in a field?